envoy/network/proxy_protocol.h

```diff
@@ -1,13 +1,25 @@
 #pragma once
 
+#include <vector>
+
 #include "envoy/network/address.h"
 
 namespace Envoy {
 namespace Network {
 
+struct ProxyProtocolTLV {
+  uint8_t type;
+  std::string key;
+  std::string value;
+};
+
+using ProxyProtocolTLVVector = std::vector<ProxyProtocolTLV>;
+using ProxyProtocolTLVVectorPtr = std::shared_ptr<std::vector<ProxyProtocolTLV>>;
+
 struct ProxyProtocolData {
   const Network::Address::InstanceConstSharedPtr src_addr_;
   const Network::Address::InstanceConstSharedPtr dst_addr_;
+  mutable ProxyProtocolTLVVectorPtr tlv_vector_ = std::make_shared<ProxyProtocolTLVVector>();
```

Comment:
[q] could this be made `const` if we parsed TLVs before initializing this?

Reply:
Yes, I believe so. I had a (wrong) impression that the options passed to transport socket was created before parsing the proxy protocol so we had to make this a mutable shared pointer, but it seems the opposite is true.